app/controllers/declaration/CommodityDetailsController.scala

```diff
@@ -56,8 +56,11 @@ class CommodityDetailsController @Inject()(
         (formWithErrors: Form[CommodityDetails]) => Future.successful(BadRequest(commodityDetailsPage(mode, itemId, formWithErrors))),
         validForm =>
           updateExportsCache(itemId, validForm).map { _ =>
-            navigator
-              .continueTo(controllers.declaration.routes.UNDangerousGoodsCodeController.displayPage(mode, itemId))
+            if (request.isType(DeclarationType.CLEARANCE)) {
+              navigator.continueTo(controllers.declaration.routes.CUSCodeController.displayPage(mode, itemId))
+            } else {
+              navigator.continueTo(controllers.declaration.routes.UNDangerousGoodsCodeController.displayPage(mode, itemId))
+            }
         }
```

Comment:
Not critical but we normally extract this logic into a sep. "nextPage" method.

Reply:
Good point, done